src/components/FollowButton/index.tsx

```diff
@@ -0,0 +1,65 @@
+import { useEffect, useState } from 'react';
+import { useRecoilValue } from 'recoil';
+
+import { useCreateFollow, useDeleteFollow } from '@/hooks/useFolllow';
+import { useModal } from '@/hooks/useModal';
+import { userAtom } from '@/recoil/user';
+import { ModalType } from '@/types/modal';
+
+import { Button } from './style';
+
+export default function FollowButton(props: any) {
+  const [isMe, setIsMe] = useState(false);
+  const [isFollowing, setIsFollowing] = useState(false);
+  const user = useRecoilValue(userAtom);
+  const { openModal } = useModal();
+
+  const follower = props.followers.filter(
+    (follower: any) => follower.user === props.userId,
+  );
+
+  const { mutate: createFollowMutate } = useCreateFollow({
+    userId: props.userId,
+  });
+
+  const { mutate: deleteFollowMutate } = useDeleteFollow();
+
+  useEffect(() => {
+    if (props.userId == user?.email) setIsMe(true);
+    if (follower.length > 0) setIsFollowing(true);
+  }, []);
+
+  const viewFriends = async () => {
+    openModal({ type: ModalType.VIEW_FOLLOW });
+  };
+
+  const follow = () => {
+    createFollowMutate(props.userId);
+    setIsFollowing(true);
+  };
+
+  const unfollow = () => {
+    deleteFollowMutate(follower[0]._id);
+    setIsFollowing(false);
+  };
```

Comment:
위의 코멘트와 이어지는 내용입니다!
follower[0]를 하게 되면 첫 번째 follower 값을 가져오게 되는데
첫 번째 follower가 내가 아닐 수도 있어서

나의 id 값으로 찾은 followData로 교체해주는게 필요해보여요!

```suggestion
  const unfollow = () => {
    if (!followData) return;
    deleteFollowMutate(followData._id);
    setFollowData(null);
  };
```